### PR TITLE
feat: update makefile with new commands

### PR DESCRIPTION
**Pull request by hughhhh**
### SUMMARY

Been working in Github Codespaces for a while now, and wanted to create new make commands to make it easier for folks to spin up s2 in the same envs. The idea here is we start to make more intelligent recipes in our makefiles that can handle failures and run the right command with devs having to figure it out on themselves

```bash
# installs all necessary requirements
$ make install

# starts postgres instance + starts flask app
$ make api 

# spins up node app
$ make client 
```

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -115,3 +115,21 @@ report-celery-beat:
...
+# Create a postgres instance using docker
+# CLI: $ psql postgresql://postgres:1234@localhost:5432/postgres
+# superset_config.py: SQLALCHEMY_DATABASE_URI = 'postgresql://postgres:1234@localhost:5432/postgres'
+db:
+	# start postgres with detached mode
+	docker run -d -p 127.0.0.1:5432:5432 -e POSTGRES_PASSWORD="1234" --name pg postgres:alpine;\
+	EXIT_CODE=$$?;\
+	echo $$EXIT_CODE;\
+	echo "command exited with $$EXIT_CODE";\
+	if [ $$EXIT_CODE -ne "0" ]; then docker start pg; echo "DB has been created"; fi
```
Comment (eschutho) [inline Makefile:128]: I usually `docker compose up db` how would spinning up a new container be different?

File: Makefile [hunk 2]
```diff
@@ -93,7 +93,7 @@ py-lint: pre-commit
 js-format:
 	cd superset-frontend; npm run prettier
 
-flask-app:
+flask-app: activate
```
Comment (eschutho) [inline Makefile:96]: what about people who use pyenv?

File: Makefile [hunk 3]
```diff
@@ -115,3 +115,21 @@ report-celery-beat:
 
 admin-user:
 	superset fab create-admin
+
+# Create a postgres instance using docker
+# CLI: $ psql postgresql://postgres:1234@localhost:5432/postgres
+# superset_config.py: SQLALCHEMY_DATABASE_URI = 'postgresql://postgres:1234@localhost:5432/postgres'
+db:
+	# start postgres with detached mode
+	docker run -d -p 127.0.0.1:5432:5432 -e POSTGRES_PASSWORD="1234" --name pg postgres:alpine;\
```
Comment (zhaoyongjie) [inline Makefile:124]: Any plaintext passwords should be avoided as much as possible, at least, providing an approach that the password pass from a command parameter.

File: Makefile (hunk 1 above)
Comment (hughhhh) [reply Makefile:128]: from my understanding docker compose is referencing a set of images in the compose yaml vs this is just one image we are targeting

this recipes are mostly focused for codespaces and has been making my workflow super easy

File: Makefile (hunk 3 above)
Comment (zhaoyongjie) [reply Makefile:124]: the makefile
```Makefile
db:
	docker run -d -p 127.0.0.1:5432:5432 -e POSTGRES_PASSWORD=$(PWD) --name pg postgres:alpine;
```

to invoke:
```shell
PWD=1234 make db
```

Comment (zhaoyongjie) [reply Makefile:124]: > even if this is for development? I'm trying to make this super easy for people to spin up superset with one command, but if you really think we should have a way to configure the password let me know

No, If you want to use it yourself, you can do it by yourself and local. If you publish it in a public codebase, please keep the basic security rules.

File: Makefile (hunk 1 above)
Comment (eschutho) [reply Makefile:128]: you can still use docker compose for a single image if you want to try it `docker compose up db` just spins up the db.

**Pull request closed without merging**